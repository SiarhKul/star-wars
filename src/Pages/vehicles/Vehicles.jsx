import "react-toastify/dist/ReactToastify.css";

import React from "react";
import { useSelector } from "react-redux";
import { ToastContainer } from "react-toastify";
import { ButtonLoadMore } from "../../components";
import { bodyCardVehicles } from "../../components/bodyCards/bodyCardVehicles";
import { Cards } from "../../components/cards/Cards";

import { GET_MORE_VEHICLES } from "../../redux/actions/actions";
import { history } from "../../redux/reducers";

export const Vehicles = () => {
	const contenCards = useSelector(state => state.dataFromServer.vehicles);
	const { isLoadedVehicles } = useSelector(state => state.loading);
	const { pathname } = history.location;

	return (
		<main>
			<ToastContainer />
			<Cards
				name="name"
				path={pathname}
				contenCards={contenCards}
				BodyComponent={bodyCardVehicles}
			/>
			{!isLoadedVehicles && <ButtonLoadMore action={GET_MORE_VEHICLES} />}
		</main>
	);
};
