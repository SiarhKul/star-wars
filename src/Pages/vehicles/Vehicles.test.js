import { mount } from "enzyme";
import React from "react";
import { Provider } from "react-redux";
import { Vehicles } from "./Vehicles";
import configureStore from "redux-mock-store";

const mockStore = configureStore();

describe("Test <Vehicles/>", () => {
	let store;
	let component;

	const getMockInitialStore = showButton => {
		return {
			dataFromServer: {
				vehicles: [],
			},
			loading: {
				isLoadedVehicles: showButton,
			},
			router: {
				location: {
					pathname: "/vehicles",
				},
			},
		};
	};

	const starships = () =>
		mount(
			<Provider store={store}>
				<Vehicles />
			</Provider>
		);

	beforeEach(() => {
		store = mockStore(getMockInitialStore(true));
		component = starships();
	});

	it("render snapshot <Vehicles/>", () => {
		expect(component).toMatchSnapshot();
	});

	it("should render ButtonLoadMore in <Vehicles/> if  all data has been loaded", () => {
		expect(component.find("ButtonLoadMore").exists()).toBeFalsy();
	});

	it("should render ButtonLoadMore in <Starships/> if  all data has not been loaded", () => {
		store = mockStore(getMockInitialStore(false));
		component = starships();
		expect(component.find("ButtonLoadMore").exists()).toBeTruthy();
	});

	it("should render alert message in <Starships/>", () => {
		expect(component.find("ToastContainer")).toHaveLength(1);
	});

	it("should render <Starships/>", () => {
		expect(component.find("Cards")).toHaveLength(1);
	});
});
