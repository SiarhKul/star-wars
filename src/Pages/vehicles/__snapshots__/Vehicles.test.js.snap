// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Vehicles/> render snapshot <Vehicles/> 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Vehicles>
    <main>
      <Loader>
        <div
          className="spinner"
        />
      </Loader>
      <ToastContainer
        autoClose={5000}
        closeButton={[Function]}
        closeOnClick={true}
        draggable={true}
        draggableDirection="x"
        draggablePercent={80}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnFocusLoss={true}
        pauseOnHover={true}
        position="top-right"
        role="alert"
        rtl={false}
        transition={[Function]}
      >
        <div
          className="Toastify"
        />
      </ToastContainer>
      <Cards
        BodyComponent={[Function]}
        contenCards={Array []}
        name="name"
        path="/"
      >
        <div
          className="cards-container"
        />
      </Cards>
    </main>
  </Vehicles>
</Provider>
`;
